x-pack/plugin/ml-package-loader/src/main/java/org/elasticsearch/xpack/ml/packageloader/action/TransportLoadTrainedModelPackage.java

```diff
@@ -106,6 +106,7 @@ protected void masterOperation(Task task, Request request, ClusterState state, A
         } catch (Exception e) {
             taskManager.unregister(downloadTask);
             listener.onFailure(e);
+            return;
```

Comment:
in general, should we always be returning after catching and passing exceptions to onFailure?

Reply:
It's a really common mistake to fire the listener and not return straight after. Otherwise the code continues and the listener may be called again leading to hard to diagnose problems.